packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -46,29 +51,85 @@ export async function convertWireToOsdkObjects(
   objects: OntologyObjectV2[],
   interfaceApiName: string | undefined,
   forceRemoveRid: boolean = false,
+  selectedProps?: ReadonlyArray<string>,
+  strictNonNull: StrictNonNull = false,
 ): Promise<Osdk<ObjectOrInterfaceDefinition>[]> {
   client.logger?.debug(`START convertWireToOsdkObjects()`);
 
-  fixObjectPropertiesInline(objects, forceRemoveRid);
+  fixObjectPropertiesInPlace(objects, forceRemoveRid);
 
-  const ret = [];
+  const ifaceDef = interfaceApiName
+    ? await client.ontologyProvider.getInterfaceDefinition(interfaceApiName)
+    : undefined;
+  const ifaceSelected = ifaceDef
+    ? (selectedProps ?? Object.keys(ifaceDef.properties))
+    : undefined;
+
+  const allResults = [];
+  const conformingResults = [];
   for (const rawObj of objects) {
     const objectDef = await client.ontologyProvider.getObjectDefinition(
       rawObj.$apiName,
     );
     invariant(objectDef, `Missing definition for '${rawObj.$apiName}'`);
 
-    if (interfaceApiName !== undefined) {
+    // default value for when we are checking an object
+    let objProps;
+
+    let conforming = true;
+    if (ifaceDef && ifaceSelected) {
       // API returns interface spt names but we cache by real values
-      reframeAsObjectInPlace(objectDef, interfaceApiName, client, rawObj);
+      invariantInterfacesAsViews(objectDef, ifaceDef.apiName, client);
+
+      conforming &&= isConforming(client, ifaceDef, rawObj, ifaceSelected);
+
+      reframeAsObjectInPlace(objectDef, ifaceDef.apiName, rawObj);
+
+      objProps = convertInterfacePropNamesToObjectPropNames(
+        objectDef,
+        ifaceDef.apiName,
+        ifaceSelected,
+      );
+    } else {
+      objProps = selectedProps ?? Object.keys(objectDef.properties);
     }
 
-    const osdkObject: any = createOsdkObject(client, objectDef, rawObj);
-    ret.push(interfaceApiName ? osdkObject.$as(interfaceApiName) : osdkObject);
+    conforming &&= isConforming(client, objectDef, rawObj, objProps);
+
+    let osdkObject = createOsdkObject(client, objectDef, rawObj);
+    if (interfaceApiName) osdkObject = osdkObject.$as(interfaceApiName);
+
+    allResults.push(osdkObject);
+    if (conforming) conformingResults.push(osdkObject);
```

Comment:
Could you maybe short circuit here for the throw case? Like if it's `throw`, as soon as you hit one non-conforming object you can just throw the error right there and not bother converting the rest of the objects?